### PR TITLE
Improved playing cards layout

### LINKED ISSUE #31: Add card graphics.
Add obverse and reverse design for each playing card in deck.

### PR DESCRIPTION
**Pull request by Kwasow**
`CardView` now includes a nice graphic showing the appropriate suit. I also added more tests and improved the interface of `GameCard`.

image

### Other changes
- Catching exceptions better
- Removed `TooGenericException` check from detekt

Closes #31

### CODE REVIEW

File: app/src/main/java/com/pokerio/app/screens/SettingsScreen.kt [hunk 1]
```diff
@@ -201,9 +212,54 @@ fun SettingsScreen(
...
+    ) {
+        Text(
+            text = stringResource(R.string.credit_arthur_camara),
+            textAlign = TextAlign.Center,
+            fontSize = 10.sp,
+            fontStyle = FontStyle.Italic,
+            color = Color.Gray
+        )
+        ClickableText(
+            text = buildLink("https://arthurcamara.co/", MaterialTheme.colorScheme.primary),
```
Comment (mikolaj-pirog) [inline app/src/main/java/com/pokerio/app/screens/SettingsScreen.kt]: I think it's better to store this url in a constant

File: app/src/main/java/com/pokerio/app/components/CardView.kt [hunk 2]
```diff
@@ -58,19 +70,51 @@ fun CardObverse(
...
+                textDecoration = numberDecoration,
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(horizontal = 4.dp)
+            )
+            Icon(
+                painter = painterResource(id = gameCard.suit.resId),
+                contentDescription = gameCard.suit.toString(),
+                modifier = Modifier
+                    .width(CARD_WIDTH * 1 / 2)
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/components/CardView.kt]: Maybe some constant for this value?

File: app/src/main/java/com/pokerio/app/utils/GameState.kt [hunk 3]
```diff
@@ -500,8 +500,8 @@ object GameState {
 
     @OptIn(ExperimentalSerializationApi::class)
     fun startGame(card1: String, card2: String, playersString: String) {
-        gameCard1 = GameCard(card1.slice(0..1), card1.slice(2..2))
-        gameCard2 = GameCard(card2.slice(0..1), card2.slice(2..2))
+        gameCard1 = GameCard.fromString(card1)
```
Comment (sormys) [inline app/src/main/java/com/pokerio/app/utils/GameState.kt:502]: GameCard.fromString throws IllegalArgumentException, shouldn't we catch and handle it?

**Review by sormys — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/pokerio/app/utils/GameState.kt (hunk 3 above)
Comment (Kwasow) [reply app/src/main/java/com/pokerio/app/utils/GameState.kt:502]: Done

File: app/src/main/java/com/pokerio/app/components/CardView.kt (hunk 2 above)
Comment (Kwasow) [reply app/src/main/java/com/pokerio/app/components/CardView.kt]: Done, also used the `size` modifier instead

File: app/src/main/java/com/pokerio/app/screens/SettingsScreen.kt (hunk 1 above)
Comment (Kwasow) [reply app/src/main/java/com/pokerio/app/screens/SettingsScreen.kt]: Stored as string resource

**Review by sormys — APPROVED** [review]

**Pull request merged**